tests/functional/src/main/java/org/mule/functional/junit4/FlowConstructRunner.java

```diff
@@ -0,0 +1,194 @@
+     * @return this {@link FlowRunner}
+     */
+    public R withInboundProperties(Map<String, Object> properties)
+    {
+        eventBuilder.withInboundProperties(properties);
+    
+        return (R) this;
+    }
+
+    /**
+     * Prepares a property with the given key and value to be sent as an outbound property of the {@link MuleMessage} to
+     * the configured flow.
+     * 
+     * @param key the key of the outbound property to add
+     * @param value the value of the outbound property to add
+     * @return this {@link FlowRunner}
+     */
+    public R withOutboundProperty(String key, Object value)
+    {
+        eventBuilder.withOutboundProperty(key, value);
+    
+        return (R) this;
+    }
+
+    /**
+     * Prepares a property with the given key and value to be sent as a session property of the {@link MuleMessage} to
+     * the configured flow.
+     * 
+     * @param key the key of the session property to add
+     * @param value the value of the session property to add
+     * @return this {@link FlowRunner}
+     */
+    public R withSessionProperty(String key, Object value)
+    {
+        eventBuilder.withSessionProperty(key, value);
+    
+        return (R) this;
+    }
+
+    /**
+     * Prepares a property with the given key and value to be sent as an invocation property of the {@link MuleMessage}
+     * to the configured flow.
+     * 
+     * @param key the key of the invocation property to add
+     * @param value the value of the invocation property to add
+     * @return this {@link FlowRunner}
+     */
+    public R withInvocationProperty(String key, Object value)
+    {
+        eventBuilder.withInvocationProperty(key, value);
+    
+        return (R) this;
+    }
+
+    /**
+     * Prepares a flow variable with the given key and value to be set in the {@link MuleMessage} to the configured
+     * flow.
+     * 
+     * @param key the key of the flow variable to put
+     * @param value the value of the flow variable to put
+     * @return this {@link FlowRunner}
+     */
+    public R withFlowVariable(String key, Object value)
+    {
+        eventBuilder.withFlowVariable(key, value);
+    
+        return (R) this;
+    }
+
+    /**
+     * Configures this runner to run this flow as one-way.
+     * 
+     * @return this {@link FlowRunner}
+     */
+    public R asynchronously()
+    {
+        eventBuilder.asynchronously();
+    
+        return (R) this;
+    }
+
+    /**
+     * Will spy the built {@link MuleMessage} and {@link MuleEvent}. See {@link Mockito#spy(Object) spy}.
+     * 
+     * @return this {@link FlowRunner}
+     */
+    public R spyObjects()
+    {
+        eventBuilder.spyObjects();
+    
+        return (R) this;
+    }
+
+    public FlowConstructRunner(MuleContext muleContext)
+    {
+        this.muleContext = muleContext;
+    }
+
+    /**
+     * Builds a new event based on this runner's config. If one has already been built, it will return that one.
```

Comment:
I don't know what are the usages for this method, but seems a bit counterintuitive that it only creates an event the first time that is invoked.